token/program-2022/src/extension/confidential_mint_burn/instruction.rs

```diff
@@ -0,0 +1,415 @@
+    /// Relative location of the
+    /// `ProofInstruction::VerifyCiphertextCommitmentEquality` instruction
+    /// to the `ConfidentialBurn` instruction in the transaction. The
+    /// `ProofInstruction::VerifyBatchedRangeProofU128` has to always be at
+    /// the instruction directly after the equality proof one,
+    /// with the `ProofInstruction::VerifyBatchedGroupedCiphertext2HandlesValidity`
+    /// following after that.
+    pub proof_instruction_offset: i8,
+}
+
+/// Create a `InitializeMint` instruction
+#[cfg(not(target_os = "solana"))]
+pub fn initialize_mint(
+    token_program_id: &Pubkey,
+    mint: &Pubkey,
+    authority: Pubkey,
+    confidential_supply_pubkey: Option<PodElGamalPubkey>,
+) -> Result<Instruction, ProgramError> {
+    check_program_account(token_program_id)?;
+    let accounts = vec![AccountMeta::new(*mint, false)];
+
+    Ok(encode_instruction(
+        token_program_id,
+        accounts,
+        TokenInstruction::ConfidentialMintBurnExtension,
+        ConfidentialMintBurnInstruction::InitializeMint,
+        &InitializeMintData {
+            authority,
+            supply_elgamal_pubkey: confidential_supply_pubkey.try_into()?,
+        },
+    ))
+}
+
+/// Create a `UpdateMint` instruction
+#[cfg(not(target_os = "solana"))]
+pub fn update_authority(
+    token_program_id: &Pubkey,
+    mint: &Pubkey,
+    authority: &Pubkey,
+    multisig_signers: &[&Pubkey],
+    new_authority: Pubkey,
+) -> Result<Instruction, ProgramError> {
+    check_program_account(token_program_id)?;
+    let mut accounts = vec![
+        AccountMeta::new(*mint, false),
+        AccountMeta::new_readonly(*authority, multisig_signers.is_empty()),
+    ];
+    for multisig_signer in multisig_signers.iter() {
+        accounts.push(AccountMeta::new_readonly(**multisig_signer, true));
+    }
+    Ok(encode_instruction(
+        token_program_id,
+        accounts,
+        TokenInstruction::ConfidentialMintBurnExtension,
+        ConfidentialMintBurnInstruction::UpdateAuthority,
+        &UpdateAuthorityData { new_authority },
+    ))
+}
+
+/// Create a `RotateSupplyElGamal` instruction
+#[allow(clippy::too_many_arguments)]
+#[cfg(not(target_os = "solana"))]
+pub fn rotate_supply_elgamal(
+    token_program_id: &Pubkey,
+    mint: &Pubkey,
+    authority: &Pubkey,
+    multisig_signers: &[&Pubkey],
+    extension_state: &ConfidentialMintBurn,
+    current_supply: u64,
+    supply_elgamal_keypair: &ElGamalKeypair,
+    new_supply_elgamal_keypair: &ElGamalKeypair,
+) -> Result<Vec<Instruction>, ProgramError> {
+    check_program_account(token_program_id)?;
+    let mut accounts = vec![
+        AccountMeta::new(*mint, false),
+        AccountMeta::new_readonly(*authority, multisig_signers.is_empty()),
+    ];
+    for multisig_signer in multisig_signers.iter() {
+        accounts.push(AccountMeta::new_readonly(**multisig_signer, true));
+    }
+
+    let new_supply_opening = PedersenOpening::new_rand();
```

Comment:
I added a `supply.rs` file to `proof-generation` ptal if there's any adjustments to be made there

Reply:
Hm, terribly sorry 🙏 , but looking at this again, it seems that it would be natural to define an `account_info` module and move the logic there since this just requires a single equality proof.

[This](https://github.com/solana-labs/solana-program-library/blob/master/token/program-2022/src/extension/confidential_transfer_fee/account_info.rs) and [this](https://github.com/solana-labs/solana-program-library/blob/master/token/program-2022/src/extension/confidential_transfer/account_info.rs) is how it is done in the other extensions.